cxx/irm.hh

```diff
@@ -49,7 +50,15 @@ class IRM {
 
   double logp_score() const;
 
-  void add_relation(const std::string& name, const T_relation& relation);
+  std::vector<Domain*> add_domains(const std::string& name,
+                                   const std::vector<std::string>& domains);
+
```

Comment:
Please add a note here if it is important that the next two add_relation's have the same name.  (For example, if you are going to invoke add_relation inside a std::visit over a T_relation.)

Reply:
I renamed one of these and added a comment for the other two.